src/main/java/uk/gov/hmcts/reform/et/syaapi/service/AssignCaseToLocalOfficeService.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.hmcts.reform.et.syaapi.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.dwp.regex.InvalidPostcodeException;
+import uk.gov.hmcts.ecm.common.model.helper.TribunalOffice;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.et.common.model.ccd.items.RespondentSumTypeItem;
+import uk.gov.hmcts.reform.et.syaapi.helper.EmployeeObjectMapper;
+import uk.gov.hmcts.reform.et.syaapi.models.CaseRequest;
+
+import java.util.List;
+
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.CASE_FIELD_MANAGING_OFFICE;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.UNASSIGNED_OFFICE;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class AssignCaseToLocalOfficeService {
+    private final PostcodeToOfficeService postcodeToOfficeService;
+
+    /**
+     * Assigns case to local office. Returns @{@link CaseData} object.
+     *
+     * @param caseRequest which would be in json format
+     * @return @link CaseData
+     */
+    public CaseData convertCaseRequestToCaseDataWithTribunalOffice(CaseRequest caseRequest) {
+        CaseData caseData = EmployeeObjectMapper.mapRequestCaseDataToCaseData(caseRequest.getCaseData());
+        List<RespondentSumTypeItem> respondentSumTypeList = caseData.getRespondentCollection();
+        String managingOffice = UNASSIGNED_OFFICE;
+        if (caseData.getClaimantWorkAddress() != null
+            && caseData.getClaimantWorkAddress().getClaimantWorkAddress() != null
+            && StringUtils.isNotBlank(caseData.getClaimantWorkAddress().getClaimantWorkAddress().getPostCode())) {
```

Comment:
Minor - maybe these if statements could be in a method - if (checkClaimantAddress), if (checkRespondentsAddress)

Reply:
fixed